### PR TITLE
Store::new(): Create path if nonexistent

### PR DESCRIPTION
**Pull request by matthiasbeyer**
`Store::new()` behaviour if the store does not exist yet.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
I consider this ready for review.

@TheNeikos , @neithernut

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -116,7 +116,17 @@ pub struct Store {
...
+
+        if !location.exists() {
+            create_dir_all(location.clone()).ok(); // TODO: Error handling?
+        } else {
+            if location.is_file() {
+                return Err(StoreError::new(StoreErrorKind::StorePathExists, None));
+            }
+        }
+
         Store {
```
Comment (TheNeikos) [inline libimagstore/src/store.rs]: Missing `Ok(`

File: libimagstore/src/store.rs [hunk 2]
```diff
@@ -116,11 +116,21 @@ pub struct Store {
 impl Store {
 
     /// Create a new Store object
-    pub fn new(location: PathBuf) -> Store {
-        Store {
+    pub fn new(location: PathBuf) -> Result<Store> {
+        use std::fs::create_dir_all;
+
+        if !location.exists() {
+            create_dir_all(location.clone()).ok(); // TODO: Error handling?
```
Comment (neithernut) [inline libimagstore/src/store.rs]: I would definitely do error handling in this case

**Comment by matthiasbeyer** [comment]
@homu retry

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
@homu r+

**Comment by matthiasbeyer** [comment]
@TheNeikos Would appreciate your review. Especially on the cargo.lock update thing... I'm not sure whether I should commit this, ... ?

**Comment by TheNeikos** [comment]
Why did you just update Cargo.lock?

**Comment by matthiasbeyer** [comment]
What do you mean by 'just'? There was noting more to update!

**Comment by TheNeikos** [comment]
Usually the lock only updates if you change the toml.

**Comment by matthiasbeyer** [comment]
Maybe it was left out at some point?

**Comment by TheNeikos** [comment]
Actually, leave it in. It won't matter as I will submit a PR that removes it since as a library one should not 
publish the lock as well.

**Comment by matthiasbeyer** [comment]
Sounds good.

**Comment by matthiasbeyer** [comment]
@TheNeikos ack?

**Comment by TheNeikos** [comment]
A Cee kay

**Pull request merged**